applications/vanilla/controllers/api/ThemesApiController.php

```diff
@@ -0,0 +1,394 @@
+            "file" => "header.html"
+        ],
+        "footer" => [
+            "type" => "html",
+            "file" => "footer.html"
+        ],
+        "variables" => [
+            "type" => "json",
+            "file" => "variables.json"
+        ],
+        "fonts" => [
+            "type" => "json",
+            "file" => "fonts.json"
+        ],
+        "scripts" => [
+            "type" => "json",
+            "file" => "scripts.json"
+        ],
+        "styles" => [
+            "type" => "css",
+            "file" => "styles.css"
+        ],
+        "inlineJs" => [
+            "type" => "js",
+            "file" => "inlineJs.js"
+        ]
+
+    ];
+
+    /* @var AddonManager */
+    private $addonManager;
+
+    /* @var ThemeModel */
+    private $themeModel;
+
+    /* @var ThemeAssetModel */
+    private $themeAssetModel;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct(
+        AddonManager $addonManager,
+        ThemeModel $themeModel,
+        ThemeAssetModel $themeAssetModel
+    ) {
+        $this->addonManager = $addonManager;
+        $this->themeModel = $themeModel;
+        $this->themeAssetModel = $themeAssetModel;
+    }
+
+    /**
+     * Get a theme assets.
+     *
+     * @param string $themeKey The unique theme key or theme ID.
+     * @return array
+     */
+    public function get(string $themeKey): array {
+
+        $in = $this->themeKeySchema('in')->setDescription('Get theme assets.');
+        $out = $this->themeResultSchema('out');
+
+        $themeAssets = $this->getThemeAssets($themeKey);
+        $themeAssets = $out->validate($themeAssets);
+        return $themeAssets;
+    }
+
+    /**
+     * Create new theme from parent theme.
+     *
+     * @param array $body Array of incoming params.
+     * @return array
+     */
+    public function post(array $body): array {
+        $this->permission("Garden.Settings.Manage");
+
+        $in = $this->themePostSchema('in')->setDescription('Post theme assets.');
+        $out = $this->themeResultSchema('out');
+        $body = $in->validate($body);
+        $themeID = $this->themeModel->insert($body);
+        if (isset($body['assets'])) {
+            foreach ($body['assets'] as $assetKey => $assetProps) {
+                if ($assetProps['type'] === 'html') {
+                    $assetRecord = [
+                        'themeID' => $themeID,
+                        'assetKey' =>$assetKey,
+                        'data' => $assetProps['body']
+                    ];
+                    $this->themeAssetModel->insert($assetRecord);
+                }
+            }
+        }
+
+        $theme = $this->getThemeAssets($themeID);
+        $themeAssets = $out->validate($theme);
+        return $themeAssets;
+    }
+
+    /**
+     * Get theme asset.
+     *
+     * @param string $id The unique theme key or theme ID (ex: keystone).
+     * @param string $assetKey Unique asset key (ex: header, footer, fonts, styles)
```

Comment:
A note here about how consumers of this API can be expected to bust the cache on the responses seems like a requirement to me.

Our `Vanilla\Web\Asset\DeploymentCacheBuster` does not seem like it would be sufficient here (due to the integration with the database). We'll likely need to create some other cache buster to go with this.